scripts/Install-Extension.ps1

```diff
@@ -0,0 +1,238 @@
+                $StopSuccess = $true
+                break;
+            }
+            Start-Sleep -Milliseconds 100
+        }
+        if (!$StopSuccess) {
+            Write-Verbose "$ServiceName failed to stop"
+        }
+    } catch {
+        Write-Verbose "Exception while waiting for $ServiceName to stop"
+    }
+
+    # Delete the service.
+    if (Get-Service $ServiceName -ErrorAction Ignore) {
+        try { sc.exe delete $ServiceName > $null }
+        catch { Write-Verbose "'sc.exe delete $ServiceName' threw exception!" }
+
+        # Wait for the service to be deleted.
+        $DeleteSuccess = $false
+        for ($i = 0; $i -lt 10; $i++) {
+            if (-not (Get-Service $ServiceName -ErrorAction Ignore)) {
+                $DeleteSuccess = $true
+                break;
+            }
+            Start-Sleep -Milliseconds 10
+        }
+        if (!$DeleteSuccess) {
+            Write-Verbose "Failed to clean up $ServiceName!"
+        }
+    }
+}
+
+<#
+.SYNOPSIS
+Starts a service with retry attempts.
+
+.PARAMETER ServiceName
+The name of the service to start.
+
+.EXAMPLE
+Start-Service-With-Retry -ServiceName "neteventebpfext"
+#>
+function Start-Service-With-Retry(
+    [Parameter(Mandatory=$true)]
+    [string]$ServiceName
+) {
+    Write-Verbose "Start-Service $ServiceName"
+    $StartSuccess = $false
+
+    for ($i=0; $i -lt 100; $i++) {
+        try {
+            Start-Sleep -Milliseconds 10
+            Start-Service $ServiceName
+            $StartSuccess = $true
+            break
+        } catch { }
+    }
+
+    if ($StartSuccess -eq $false) {
+        Write-Error "Failed to start $ServiceName"
+    }
+}
+
+<#
+.SYNOPSIS
+Installs a service and starts it.
+
+.PARAMETER ServiceName
+The name of the service to install.
+
+.PARAMETER BinaryPath
+The path to the binary to install as a service. This MUST be the full path.
+
+.EXAMPLE
+Install-Service -ServiceName "neteventebpfext" -BinaryPath "C:\neteventebpfext.sys"
+#>
+function Install-Service(
+    [Parameter(Mandatory=$true)]
+    [string]$ServiceName,
+    [Parameter(Mandatory=$true)]
+    [string]$BinaryPath
+)
+{
+    # Cleanup service if it already exists.
+    $ServiceExists = Get-Service -Name $ServiceName -ErrorAction SilentlyContinue
+    if ($ServiceExists) {
+        Write-Verbose "$ServiceName already exists. Attempting to cleanup the service first."
+        Clear-Service -ServiceName $ServiceName
+    }
+
+    # Install the service.
+    sc.exe create $ServiceName type= kernel binpath= $BinaryPath start= system | Write-Verbose
+    if ($LastExitCode) { Write-Error "Failed to install driver" }
+
+    # Start the service.
+    Start-Service-With-Retry $ServiceName
+
+    Write-Verbose "$ServiceName install complete!"
+}
+
+<#
+.SYNOPSIS
+This function sets up paths and names needed for installing/uninstalling neteventebpfext
+#>
+function Set-Path-Neteventebpfext {
+    $script:ExtPath = $BinaryDirectory + "\neteventebpfext.sys"
```

Comment:
Consider using Join-Path instead of + operator. Join-Path should help ensure there is no extra '\' (in case the caller passes one in)